### PR TITLE
Changes to scalar (normal and biharmonic) diffusivity

### LINKED ISSUE #2264: `Explicit`/`Implicit` names for time-discretizations are a bit vague?
The recent diffusivity refactor #2247 changed `ExplicitTimeDiscretization` -> `Explicit`.

https://github.com/CliMA/Oceananigans.jl/blob/b3ddbc84c8f35aaf5b93fbbfdb4cffcada5c6533/src/TurbulenceClosures/implicit_explicit_time_discretization.jl#L8-L27

We are importing these in http://github.com/CliMA/OceanTurbulenceParameterEstimation.jl and it seems a bit vague to import

```Julia
using Oceananigans.TurbulenceClosures: Explicit
```

Should we go back to bit more verbose names?

cc: @simone-silvestri, @glwagner

### PR DESCRIPTION
**Pull request by simone-silvestri**
closes #2266 
closes #2264
closes https://github.com/CliMA/Oceananigans.jl/issues/2261

### CODE REVIEW

**Review by glwagner — APPROVED** [review]

File: validation/mesoscale_turbulence/baroclinic_adjustment.jl [hunk 1]
```diff
@@ -8,7 +8,7 @@ using JLD2
 using Oceananigans
 using Oceananigans.Units
 using Oceananigans.Models.HydrostaticFreeSurfaceModels: fields
-using Oceananigans.TurbulenceClosures: VerticallyImplicit, Vertical, Horizontal
+using Oceananigans.TurbulenceClosures
```
Comment (tomchor) [inline validation/mesoscale_turbulence/baroclinic_adjustment.jl:11]: Can we delete this line?

File: validation/mesoscale_turbulence/eddying_channel_nonhydrostatic.jl [hunk 2]
```diff
@@ -9,7 +9,7 @@ using Oceananigans.Diagnostics
 using Oceananigans.Utils
 using Oceananigans.AbstractOperations
 using Oceananigans.Advection
-using Oceananigans.TurbulenceClosures: Vertical, Horizontal
+using Oceananigans.TurbulenceClosures
```
Comment (tomchor) [inline validation/mesoscale_turbulence/eddying_channel_nonhydrostatic.jl:12]: Can we also delete this line (I think there are more that remain)

Comment (navidcy) [reply validation/mesoscale_turbulence/eddying_channel_nonhydrostatic.jl:12]: Yeap

File: validation/mesoscale_turbulence/baroclinic_adjustment.jl (hunk 1 above)
Comment (simone-silvestri) [reply validation/mesoscale_turbulence/baroclinic_adjustment.jl:11]: I explicitly exported 
```
    ExplicitTimeDiscretization,
    VerticallyImplicitTimeDiscretization,
    ThreeDimensionalFormulation,
    HoriozontalFormulation,
    VerticalFormulation
```

in the module `TurbulenceClosures`

**Comment by tomchor** [comment]
@simone-silvestri thanks a lot for the help here. I added two commits that mostly clarify/fix docs that I migrated from https://github.com/CliMA/Oceananigans.jl/pull/2266 (which I will now close),  I hope you don't mind.

**Comment by simone-silvestri** [comment]
@tomchor no problem, indeed change stuff where you see fit. By the way, I didn't understand if we want to have all the types exported in `Oceananingans` or only in `TurbulenceClosures`. I would leave them in `TurbulenceClosures`

**Comment by tomchor** [comment]
> @tomchor no problem, indeed change stuff where you see fit. By the way, I didn't understand if we want to have all the types exported in `Oceananingans` or only in `TurbulenceClosures`. I would leave them in `TurbulenceClosures`

I think now that we have aliases (`VerticalScalarDiffusivity` and `HorizontalScalarDiffusivity`) exporting them at the top isn't necessary anymore. But that's just my opinion.

**Comment by simone-silvestri** [comment]
> > @tomchor no problem, indeed change stuff where you see fit. By the way, I didn't understand if we want to have all the types exported in `Oceananingans` or only in `TurbulenceClosures`. I would leave them in `TurbulenceClosures`
> 
> I think now that we have aliases (`VerticalScalarDiffusivity` and `HorizontalScalarDiffusivity`) exporting them at the top isn't necessary anymore. But that's just my opinion.

Agreed! I ll remove them

**Review by tomchor — APPROVED** [review]
LGTM!

note: I added a line to close https://github.com/CliMA/Oceananigans.jl/issues/2261 in the PR description.

**Comment by simone-silvestri** [comment]
Sure, tests should pass in this last commit, when that is done I ll merge!

**Comment by navidcy** [comment]
Wasn't some of the changes introduced by this PR breaking ones?

**Pull request merged**